src/Chainweb/BlockHeader.hs

```diff
@@ -286,6 +297,76 @@ encodeBlockCreationTime (BlockCreationTime t) = encodeTime t
 decodeBlockCreationTime :: MonadGet m => m BlockCreationTime
 decodeBlockCreationTime = BlockCreationTime <$> decodeTime
 
+-- -------------------------------------------------------------------------- --
+-- POW Target Computation
+
+newtype EpochStartTime = EpochStartTime (Time Micros)
+    deriving stock (Show, Eq, Ord, Generic)
+    deriving anyclass (NFData)
+    deriving newtype (ToJSON, FromJSON, Hashable, LeftTorsor)
+
+instance IsMerkleLogEntry ChainwebHashTag EpochStartTime where
+    type Tag EpochStartTime = 'EpochStartTimeTag
+    toMerkleNode = encodeMerkleInputNode encodeEpochStartTime
+    fromMerkleNode = decodeMerkleInputNode decodeEpochStartTime
+    {-# INLINE toMerkleNode #-}
+    {-# INLINE fromMerkleNode #-}
+
+encodeEpochStartTime :: MonadPut m => EpochStartTime -> m ()
+encodeEpochStartTime (EpochStartTime t) = encodeTime t
+
+decodeEpochStartTime :: MonadGet m => m EpochStartTime
+decodeEpochStartTime = EpochStartTime <$> decodeTime
+
+isEpochStartHeight :: HasChainwebVersion v => v -> BlockHeight -> Bool
+isEpochStartHeight v h = case window (_chainwebVersion v) of
+    Nothing -> False
+    Just (WindowWidth n)
+        | h == 0 -> True
+        | int h < n && int h `mod` 10 == (0 :: Int) -> True
+        | int h `mod` n == 0 -> True
+        | otherwise -> False
+{-# INLINE isEpochStartHeight #-}
+
+isLastInEpoche :: BlockHeader -> Bool
```

Comment:
`s/Epoche/Epoch`

Reply:
Oh je.

Reply:
Ahhh. Is there a way to configure my keyboard (on hardware level) to prevent me from writing "epoche"?